usage/usage.py

```diff
@@ -372,25 +372,35 @@ def get_first_last(jobs, key_len, mapping):
     return start_end
 
 
-def get_nodehours(times, nodes):
-    """Takes the times and nodes arrays and outputs the node-hours used for
+def get_nodehours(data):
+    """Takes the accounting data from SLURM and outputs the node-hours used for
     all of the slurm jobs requested.
 
     Parameters
     ----------
-    times : Numpy array of `float`
-        time array from the data file in hours
-    nodes : Numpy array of `int`
-        array of node usage from the data file
+    data : `list` of `dict`
+        accounting data from slurm.
 
     Returns
     -------
     node_hours : `float`
         the total node-hours spent on all of the slurm jobs passed into
         usage.py.
     """
-    dt = times[1] - times[0]
-    node_hours = np.sum(dt*nodes)
+    # Find duration of each job
+    duration = [(rec['end'] - rec['start']).total_seconds() for rec in data]
+    nodes = [int(rec['nnodes']) for rec in data]
+
+    # Collect elapsed time data
+    node_hours = 0
+    for dur, node in zip(duration, nodes):
+        node_hours += (dur*node)/3600.0
+
+    sigfig = (len((str(int(node_hours*3600))).replace(".", "")))
+
+    # First you add, then you found to sig figs.
+    node_hours = float(('{0:.%ig}' % sigfig).format(node_hours))
+
```

Comment:
I must admit I'm slightly confused about the changes. I mean I understand your goal, not the execution though.

Let's focus on lines 395-404. Considering usual processing times, rounding up number of hours to one (or two) digits after decimal point would be probably more than enough. Then the fragment in question could be written as simple as this:
```python
total = sum(dur*node for dur, node in zip(duration, nodes))
return round(total/3600.0, 2)
```
So what am I missing?

Reply:
You have a good point; I think I was just accidentally over complicating things.